src/llama.cpp

```diff
@@ -5530,6 +5530,9 @@ static void llm_load_vocab(
                 vocab.tokenizer_clean_spaces = false;
                 vocab.tokenizer_ignore_merges = true;
                 vocab.tokenizer_add_bos = true;
+            } else if (
+                tokenizer_pre == "smollm") {
+                vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_SMOLLM;
```

Comment:
```suggestion
                vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_SMOLLM;
                vocab.tokenizer_clean_spaces = false;
```

See <https://huggingface.co/HuggingFaceTB/SmolLM-135M/blob/0ccfeb9654bc6ae54109cf2a4bb6277921a5c365/tokenizer_config.json#L161>

Reply:
Thank you @compilade